File: tests/fixtures/go/method_calls.go
package main

type Processor struct {
	Data string
}

func NewProcessor(data string) *Processor {
	return &Processor{Data: data}
}

func (p *Processor) Process() string {
	return p.Data
}

func (p *Processor) Validate(input string) string {
	return input
}

func run() {
	p := NewProcessor("test")
	result := p.Process()
	valid := p.Validate(result)
	_ = valid
}

File: tests/fixtures/rust/method_calls.rs
struct Processor {
    data: String,
}

impl Processor {
    fn new(data: String) -> Self {
        Processor { data }
    }

    fn process(&self) -> &str {
        &self.data
    }

    fn validate(&self, input: &str) -> &str {
        input
    }
}

fn run() {
    let p = Processor::new(String::from("test"));
    let result = p.process();
    let valid = p.validate(result);
    let _ = valid;
}
